### PR TITLE
Adding docs and website folder required for `gh-pages`

### PR DESCRIPTION
**Pull request by pratap-akhand**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build was run locally and any changes were pushed
- [ ] Lint has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
We are using `docusaurus` tool to created documentation for couchbase plugin. Here we are adding the two folders:
docs: It contains all `.md` files
website: This contains some changes regarding webpage creation and conversion from .md to .html format.
There is one more review raised for the branch `gh-pages` for this repo, where actual build/pages is deployed. In this PR adding the input resource code/files required to generate the build.

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: website/siteConfig.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    {doc: 'Overview', label: 'Docs'},
+    {page: 'help', label: 'Help'},
+    { search: true },
+  ],
+
+  // If you have users set above, you add it here:
+
+
+  /* path to images for header/footer */
+  headerIcon: '/img/delphix-logo-white.png',
```
Comment (fdrozdowski) [inline website/siteConfig.js:45]: For some reason the delphix logo looks stretched in the header (https://pratap-akhand.github.io/couchbase-plugin/docs/Overview).

File: docs/BuildPlugin.md [hunk 2]
```diff
@@ -1,44 +1,48 @@
...
-
-```dvp download-logs -c plugin_config.yml -e <Delphix_Engine_Name> -u admin --password <password>```
-
-#### Unit test logs: 
-##### SummaryReport:
-A report with name `Report.html` gets generated in `test` directory which contains the summary of test passed vs failed. If any test case fails then complete stack trace can be seen in that test case section.
-##### Module wise coverage report:
-There is a report folder `CodeCoverage`(can change the folder name in config file `pytest.ini`) generated in `test` directory, which contains html files. These files help in source code coverage visualization, in which we can see statements processed and missed in each module of source code.
+---
+id: BuildPlugin
```
[the remark this answers is not in the document]
Comment (pratap-akhand) [reply docs/BuildPlugin.md]: Fixed.

File: docs/BuildPlugin.md [hunk 3]
```diff
@@ -1,44 +1,48 @@
...
+     pytest test/
+```
+
+### <a id="run_unit_test_case"></a>Download plugin logs
+#### Plugin Logs:
+Download the plugin logs using below command:
+
+```dvp download-logs -c plugin_config.yml -e <Delphix_Engine_Name> -u admin --password <password>```
+
+#### Unit test logs:
```
[the remark this answers is not in the document]
Comment (pratap-akhand) [reply docs/BuildPlugin.md]: Fixed

File: docs/InstallPlugin.md [hunk 4]
```diff
@@ -1,4 +1,7 @@
-#  Install Couchbase Plugin
+---
+id: InstallPlugin
+title: Install Couchbase Plugin
```
[the remark this answers is not in the document]
Comment (pratap-akhand) [reply docs/InstallPlugin.md]: Fixed

File: docs/InstallPlugin.md [hunk 5]
```diff
@@ -1,4 +1,7 @@
-#  Install Couchbase Plugin
+---
+id: InstallPlugin
```
[the remark this answers is not in the document]
Comment (pratap-akhand) [reply docs/InstallPlugin.md]: Fixed

File: docs/Overview.md [hunk 6]
```diff
@@ -1,5 +1,7 @@
-# Overview
-
+---
+id: Overview
+title: Couchbase-plugin Overview
```
[the remark this answers is not in the document]
Comment (pratap-akhand) [reply docs/Overview.md]: Fixed

File: docs/Overview.md [hunk 7]
```diff
@@ -1,5 +1,7 @@
...
+---
 
 Couchbase Plugin
 ----------------
```
[the remark this answers is not in the document]
Comment (pratap-akhand) [reply docs/Overview.md:7]: Fixed

File: website/siteConfig.js (hunk 1 above)
Comment (pratap-akhand) [reply website/siteConfig.js]: Fixed

**Pull request closed without merging**